doc/rst/technotes/globalvars.rst

```diff
@@ -0,0 +1,47 @@
+.. _readme-globalvars:
+
+==================================
+Global Variables
+==================================
+
+.. warning:: The variables defined here are not part of Chapel Standard and 
+             they can be changed, renamed, or removed at any point in the future.
+
+The Chapel Compiler defines some variables that are available to all Chapel programs. 
+These variables are used by the compiler itself and the standard Chapel libraries/packages 
+to provide some features/optimizations based on compilation flags. 
+All of these variables are compile-time constants.
+
+Variables
+---------
+
+**boundsChecking**
+
+    This variable can be used to check if ``bounds-checks`` are enabled or not. 
+    It is used by libraries to enable/disable runtime bound checks while 
+    accessing ranges, arrays, or domains.
+
+**castChecking**
+
+    This variable reflects whether ``cast-checks`` are enabled or not.
+    It is used to provide checks before casting a variable from one type to another.
```

Comment:
In reviewing the use of this variable, it seems to apply in very limited circumstances:

`integral.safeCast()`, and `channel`'s `readbits()` and `writebits()` methods(?).  So it doesn't apply to most casts.

Reply:
All these casts apply to situation where there is a chance of losing information or narrowing values, should this be mentioned explicitly?

Reply:
> All these casts apply to situation where there is a chance of losing information or narrowing values, should this be mentioned explicitly?

That seems reasonable to me.

> In reviewing the use of this variable, it seems to apply in very limited circumstances

I think "in some cases" would be enough wiggle words to indicate that it isn't necessarily applied everywhere that it could be.  It also means that its use can be expanded without requiring us to be constantly updating this documentation